### DISCUSSION TITLE
Option to use radians with trigonometric functions

### DISCUSSION THREAD

**Discussion by afontenot** [libvips/libvips#2086] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm new to using vips, and I got really tripped up over vips' trigonometric functions taking and returning degrees as arguments instead of radians. This seems really non-standard to me. Plenty of math (e.g. the arc length formula) requires you to work with radians, so you end up converting back.

Worse still I started doing some digging, and it seems from [math.c#L134](https://github.com/libvips/libvips/blob/master/libvips/arithmetic/math.c#L134) and [util.h#L51](https://github.com/libvips/libvips/blob/master/libvips/include/vips/util.h#L51) that vips is actually *converting* the inputs (or outputs for the inverse functions) to/from radians since that's what the underlying functions use anyway. This results in an unnecessary loss of precision, which is doubled if I'm converting to/from degrees on my end.

**Describe the solution you'd like**
Obviously the cleanest solution would be to change the trigonometric functions so that radians are used by default. I'm assuming this can't happen because of backwards compatibility. The best alternative seems to be adding a parameter to use radians, and having its default (unspecified) value be to use degrees.

**Describe alternatives you've considered**
Alternative functions could be added with different names, e.g. `sin_radians`.

**Comment by jcupitt** [comment]
Hi @afontenot, sure, a bool argument like `radians` to enable radians mode would be easy. It would be slightly annoying to use though I suppose. There could also be an environment variable, like `VIPS_RADIANS`, or perhaps a global setting you change with `vips_set_radians(TRUE)`.

libvips is degrees by default because of its background in colour science. All the trig colour spaces (eg. LCh) are in degrees and mixing degrees and radians in one program always causes intense confusion.

Let's leave this open as an enhancement.

**Comment by jcupitt** [comment]
Thinking about this again, I'm less sure. What about complex numbers? We'd need a radians mode for polar and rectangular conversions there too. And then when they interacted with the colour functions, we'd need to add conversions, eg. `vips_rad` and `vips_deg` operations to swap between them. The transforms like `vips_rotate()` work in degrees too -- should they have a radians mode? This sounds like a recipe for annoying bugs.

Although it maybe looks like a strange choice if you come from something like mathematica, it is more normal in image processing, and especially in colour science. I think the simplicity of using degrees everywhere is probably better than the confusion you'd get with two units for angles.

I'm inclined to close as wontfix, sorry.

**Comment by afontenot** [comment]
> Although it maybe looks like a strange choice if you come from something like mathematica, it is more normal in image processing, and especially in colour science. I think the simplicity of using degrees everywhere is probably better than the confusion you'd get with two units for angles.

Yep, I can understand the rational for doing things differently, depending on the field. If vips were being newly written from scratch I think I'd push a bit harder to make radians the default, but since using degrees is already the norm with this project maybe it's not worth it. Feel free to close if you want.
